4/README.md

```diff
@@ -0,0 +1,346 @@
+
+```abnf
+rel-change = change
+abs-change = chagne
+
+archive-rel-change = change
+archive-abs-change = chagne
+
+number = [ "-" ] 1*DIGIT [ "." ] *DIGIT
+change = number [ "," number ]
+
+period = [ "-" ] 1*DIGIT
+archive-period = period
+```
+
+### Attribute runtime parameters
+
+At given point in time an Attribute MUST have associated:
+* *quality*, an enumeration describing the state *read value*
+  (one of *VALID*, *INVALID*, *ALARM*, *CHANGING*, *WARNING*),
+* *read value*, an object representing the value of the Attribute.
+  It MUST conform to *data format* and *data type*,
+* *read dim x*, an integer describing the number of data elements in
+  *read value* in X dimension.
+  If *data format* is *SCALAR*, it MUST be either 0 or 1.
+  If *data format* is *SPECTRUM* or *IMAGE*, it MUST be between 0 and
+  *max dim x*.
+* *read dim y*, an integer describing the number of data elements in
+  *read value* in Y dimension.
+  If *data format* is *SCALAR* or *SPECTRUM*, it MUST be 0.
+  If *data format* is *IMAGE*, it MUST be between 0 and *max dim y*.
+
+Additionally, if *writable* is *WRITE* or *READ_WRITE*,
+at given point in time an Attribute MUST have associated:
+* *set value*, an object representing the value set to the Attribute.
+  It MUST conform to *data format* and *data type*,
+* *write dim x*, an integer describing the number of data elements in
+  *set value* in X dimension,
+* *write dim y*, an integer describing the number of data elements in
+  *set value* in Y dimension.
+
+
+### Attribute aliases
+
+A *full attribute name* consists of *device name* and *attribute name*
+separated by "/" character.
+
+See [RFC 2/Device](/2) for the definition of `<device-name>`.
+
+An Attribute MAY have associated *alias*, a string which can be used
+in place of *full attribute name* to address the Attribute.
+An Attribute MUST NOT have more than one *alias*.
+
+An *alias* MUST be stored in the database.
+
+An *alias* MUST be unique.
+
+An *alias* MAY contain any character (including multi-byte characters) except
+that it MUST NOT contain any of `/`, ` ` (space), `#`, `:`, `->`.
+It MAY be empty.
+See `<attribute-alias>` specification below.
+
+
+### Attribute events
+
+An Attribute can send following events:
+* *PERIODIC*,
+* *CHANGE*,
+* *ARCHIVE*,
+* *PIPE*,
+* *DATA READY*,
+* *ATTR CONF*,
+* *USER*,
+
+In order to send *PERIODIC*, *CHANGE*, *ARCHIVE*, *PIPE* and *DATA READY*
+events, an attribute must be polled. See
+[Attribute polling section](#attribute-events) for more details.
+
+See [Attribute properties section](#attribute-properties) for specific
+conditions under which the events are sent.
+
+
+### Attribute polling
+
+An Attribute can be polled. A polling for the Attribute can be configured
+in Device with *polled attr* Property. See [RFC XX/Cache system](/XX) for
+more details.
+
+
+### Attribute naming schema
+
+Formal specification of Attribute name is given below:
+```ABNF
+attribute-name = *attribute-name-char
```

Comment:
I think we should impose at least one character here because it does not work properly in the current system with empty attribute names.

Reply:
Do you mean that it will be rejected by Pogo or will not work with tools like Jive?
I already added a note:
"Also note that the Attribute name SHOULD contain at least one letter and SHOULD NOT start with a digit."

In my opinion the proper place to put any restrictions on attribute name would be constructor `Attr::Attr`. Otherwise we will have a specification which is not followed by any implementation.

Reply:
> Do you mean that it will be rejected by Pogo or will not work with tools like Jive?

I mean there will be some unexpected side effects for sure. I didn't try it so I don't know how bad it will behave actually.

> I already added a note:
> "Also note that the Attribute name SHOULD contain at least one letter and SHOULD NOT start with a digit."

Indeed, I saw this. The abnf definition is much more relaxed. Maybe we should add a recommended attribute name abnf definition? I understand your point of view. You're simply documenting the current implementation.
I think we should not tolerate empty attribute names in next major Tango versions.

> 
> In my opinion the proper place to put any restrictions on attribute name would be constructor `Attr::Attr`. 

I agree with you